gosu-core-api/src/main/gosu/gw/lang/enhancements/CoreIterableEnhancement.gsx

```diff
@@ -557,4 +558,25 @@ enhancement CoreIterableEnhancement<T> : java.lang.Iterable<T> {
     }
     return retList
   }
-}
\ No newline at end of file
+
+  /**
+   * takes two lists and returns a list of corresponding <code>gw.util.Pair</code>s.
+   * If one input list is short, excess elements of the longer list are discarded.
+   */
+  function zip<R>( other : Iterable<R>) : Iterable<Pair<T,R>> {
+    if (other == null) {
+      throw new NullPointerException("other should be non-null")
+    }
+
+    var zippedLength = this.Count < other.Count ? this.Count : other.Count
+    var zipped = new ArrayList<Pair<T,R>>(zippedLength)
+
```

Comment:
I didn't think of using a LinkedList to be honest. I mean, even though it's unlikely, if someone tried to append an element to the returned Iterable<Pair<T,R>> then you'd still get the dynamic allocation issue so it feels right to use a LinkedList to make that go away. I'll make the change.